### DISCUSSION TITLE
Linter error with bits/stdc++.h header file

### DISCUSSION THREAD

**Discussion by mrmagicpotato007** [cpeditor/cpeditor#824] [Q&A] [answered]
Hi i've downloaded llvm and added clang.exe to the path.I'm facing linting for #include<bits/stdc++.h> as well as include<vector>.how can i resolve this issue?

**Comment by coder3101** [comment]
The best way to fix it is to not use it.

[From GeeksForGeeks](https://www.geeksforgeeks.org/bitsstdc-h-c/)

> bits/stdc++.h is not a standard header file of GNU C++ library. So, if you try to compile your code with some compiler other than GCC it might fail; e.g. MSVC and CLANG do not have this header.

This explains why you get linter error, because `clangd` uses clang++ headers to provide linting since it cannot find it. It flags as error.


> This header file is not part of the C++ standard and is therefore, non-portable, and should be avoided.

This is the perfect solution for your error.

However if you don't really want to type in a lot of header files everytime you start a new c++ tab for coding. You can [use a template](https://cpeditor.org/docs/preferences/language/#template) with all your needed headers included.

**Comment by ouuan** [comment]
You can try downloading MinGW and LLVM at https://github.com/brechtsanders/winlibs_mingw/releases.

**Comment by ouuan** [comment, marked as answer]
Since CP Editor 6.9.2, using the `clangd` in the "with-gcc-x.x.x-LLVM-x.x.x" release should solve this problem.

**Reply by saidurraahmaan** [reply]
I have previously installed LLVM .  Should  I uninstall it?

**Reply by coder3101** [reply]
>  I have previously installed LLVM .  Should  I uninstall it?

Yes and use the LLVM that is packaged with cpeditor since beta.
